src/inject/index.ts

```diff
@@ -47,11 +53,11 @@ const colorSchemeWatcher = watchForColorSchemeChange(({isDark}) => {
     chrome.runtime.sendMessage({type: 'color-scheme-change', data: {isDark}});
 });
 
-const port = chrome.runtime.connect({name: 'tab'});
-port.onMessage.addListener(onMessage);
-port.onDisconnect.addListener(() => {
-    logWarn('disconnect');
-    cleanDynamicThemeCache();
-    colorSchemeWatcher.disconnect();
-});
+chrome.runtime.onMessage.addListener(onMessage);
+chrome.runtime.sendMessage({type: 'tab'});
+
+function onUnload() {
+    chrome.runtime.sendMessage({type: 'tab-unload'});
+}
 
+addEventListener('unload', onUnload);
```

Comment:
As I just said I don't like to use of `unload` event in anyway as it's unpredictable and kind of a black magic use case within browsers(especially cross-versions). And I cannot accept any code with this kind of event because browsers will not use [back forward cache](https://developers.google.com/web/updates/2018/07/page-lifecycle-api#back-forward-cache) which I know will create a tsunami of issues opening about this behavior from dark reader. Also see: https://developers.google.com/web/updates/2018/07/page-lifecycle-api#the-beforeunload-event.